packages/core/src/components/block/block.stories.tsx

```diff
@@ -49,31 +53,49 @@ Default.argTypes = {
       defaultValue: { summary: 'Inherit from parent' },
     },
   },
+  componentTag: {
+    name: 'Component Tag',
+    description: 'Specifies the HTML tag used for the component wrapper.',
+    control: {
+      type: 'radio',
+    },
+    options: ['section', 'div', 'article', 'aside', 'header', 'footer', 'nav', 'main'],
+    table: {
+      defaultValue: { summary: 'section' },
+    },
+  },
 };
 
 Default.args = {
   modeVariant: 'Inherit from parent',
+  componentTag: 'section',
 };
 
-const NestedTemplate = ({ outerModeVariant }) =>
+const NestedTemplate = ({ outerModeVariant, componentTag }) =>
   formatHtmlPreview(
     `
       <div ${
         outerModeVariant !== 'Inherit from parent'
           ? `class="tds-mode-variant-${outerModeVariant.toLowerCase()}"`
           : ''
       }>
-        <tds-block>
-          <h2 class="tds-headline-02">Outer Block</h2>
-          <p class="tds-body-01">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
-          <tds-block>
-            <h3 class="tds-headline-04">Middle Block</h3>
-            <p class="tds-detail-03">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
-            <tds-block>
-              <h4 class="tds-headline-06">Inner Block</h4>
-              <p class="tds-detail-03">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
+        <tds-block component-tag="${componentTag}">
+          <${componentTag}>
+            <h2 class="tds-headline-02">Outer Block (${componentTag})</h2>
+            <p class="tds-body-01">This block is now structured using a <code>&lt;${componentTag}&gt;</code>.</p>
+            <tds-block component-tag="aside">
+              <aside>
+                <h3 class="tds-headline-04">Middle Block (Aside)</h3>
+                <p class="tds-detail-03">Nested content inside an <code>&lt;aside&gt;</code> element.</p>
+                <tds-block component-tag="section">
+                  <section>
+                    <h4 class="tds-headline-06">Inner Block (Section)</h4>
+                    <p class="tds-detail-03">Ensuring meaningful content structure with semantic HTML.</p>
+                  </section>
+                </tds-block>
+              </aside>
             </tds-block>
-          </tds-block>
+          </${componentTag}>
```

Comment:
Why is the componentTag needed to close out the template here? Should this just be `tds-block` ?